presto-hive/src/test/java/io/prestosql/plugin/hive/AbstractTestHive.java

```diff
@@ -2556,6 +2557,38 @@ public void testViewCreation()
         }
     }
 
+    @Test
+    public void testHiveViewTranslationError()
+    {
+        String tableName = "test_hive_view";
+        ConnectorSession session = newSession();
+        Table table = Table.builder()
+                .setDatabaseName(database)
+                .setTableName(tableName)
+                .setOwner(session.getUser())
+                // Create a view without using PRESTO_VIEW_FLAG
+                .setTableType(TableType.VIRTUAL_VIEW.name())
+                .setDataColumns(ImmutableList.of(new Column("dummy", HIVE_STRING, Optional.empty())))
+                .setPartitionColumns(ImmutableList.of())
+                .setViewOriginalText(Optional.of("SELECT 'a'"))
+                .withStorage(storage -> storage.setStorageFormat(VIEW_STORAGE_FORMAT)
+                        .setLocation("")
+                        .build())
+                .build();
+
+        metastoreClient.createTable(new HiveIdentity(session), table, testingPrincipalPrivilege(session));
+
+        try (Transaction transaction = newTransaction()) {
+            ConnectorMetadata metadata = transaction.getMetadata();
+            assertThatThrownBy(() -> metadata.getView(session, new SchemaTableName(database, tableName)))
+                    .isInstanceOf(HiveViewNotSupportedException.class)
+                    .hasMessageContaining("Hive views are not supported");
```

Comment:
It would be nice add a test where changing config property would actually return a view here.

We have product tests for that, so if adding a test here is problematic then maybe a comment would be enough.